compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.cpp

```diff
@@ -509,15 +509,25 @@ getMatmulOrIGEMMLoweringConfigAndWorkgroupSize(
   attrs.emplace_back(StringAttr::get(context, "mma_kind"), kind);
   if (mustBeAligned) {
     Attribute useGlobalDma = IREE::GPU::UseGlobalLoadDMAAttr::get(context);
-    Attribute promotionArray[] = {useGlobalDma, useGlobalDma};
+    SmallVector<Attribute> promotionArray = {useGlobalDma, useGlobalDma};
+    SmallVector<int64_t> promotionList = {0, 1};
+    if (scaled) {
+      promotionArray.append({useGlobalDma, useGlobalDma});
+      promotionList.append({2, 3});
+    }
     ArrayRef<Attribute> promotionTypes =
         useDirectLoad ? ArrayRef<Attribute>(promotionArray)
                       : ArrayRef<Attribute>{};
-    GPU::appendPromotedOperandsList(context, attrs, {0, 1}, promotionTypes);
+    GPU::appendPromotedOperandsList(context, attrs, promotionList,
+                                    promotionTypes);
   } else {
     // TODO (nirvedhmeshram, Max191, jerryyin) : Add support so that unaligned
     // shapes do not require c promotion.
-    GPU::appendPromotedOperandsList(context, attrs, {0, 1, 2});
+    SmallVector<int64_t> promotionList = {0, 1, 2};
+    if (scaled) {
+      promotionList.push_back(3);
```

Comment:
Since this is doing C promotion, you want append_range of  `{3, 4}` here

But also